engines/sci/parser/vocabulary.cpp

```diff
@@ -250,10 +251,38 @@ bool Vocabulary::loadSuffixes() {
 
 		_parserSuffixes.push_back(suffix);
 	}
+	appendSuffixes();
 
 	return true;
 }
 
+void Vocabulary::appendSuffixes() {
+	if (g_sci->getLanguage() == Common::HE_ISR) {
+		for (int i = 0; i < 2; i++) {
+			int cls;
+			if (i == 0)
+				cls = VOCAB_CLASS_PREPOSITION << 4;
+			else
+				cls = VOCAB_CLASS_NOUN << 4;
+
+			suffix_t suffix1 = { cls, cls, 1, 0, "\xea", "" };					// get rid of Kaf Sofit
+			_parserSuffixes.push_back(suffix1);
+
+			suffix_t suffix2 = { cls, cls, 2, 0, "\xe9\xed", "" };				// get rid of Yud, Mem Sofit
+			_parserSuffixes.push_back(suffix2);
+
+			suffix_t suffix3 = { cls, cls, 2, 0, "\xe5\xfa", "" };				// get rid of Vav, Taf
+			_parserSuffixes.push_back(suffix3);
+
+			suffix_t suffix4 = { cls, cls, 3, 2, "\xe9\xe5\xfa", "\xe9\xfa" };	// Yud, Vav, Taf -> Yud, Taf
+			_parserSuffixes.push_back(suffix4);
+
+			suffix_t suffix5 = { cls, cls, 3, 2, "\xe0\xe5\xfa", "\xe0\xe4" };	// Alef, Vav, Taf -> Alef, He
+			_parserSuffixes.push_back(suffix5);
```

Comment:
You can put all of these prefixes into an array, and iterate through it, instead of repeating the same code